46.md

```diff
@@ -1,98 +1,162 @@
+
+Coming soon...
+
+### Example Oauth-like flow to create a new user account with Nsecbunker
+
+Coming soon...
+
+## Request Events `kind: 24133`
+
+```json
 {
-  "pubkey": "<client-key-hex>"
-  "kind": 24133,
-  "tags": [
-    ["p", "<signer-key-hex>"]
-  ],
-  "content": "nip04_encrypted_json({id: <random-string>, method: <see-below>, params: [array_of_strings]})",
-  ...
+    "id": <id>,
+    "kind": 24133,
+    "pubkey": <local_keypair_pubkey>,
+    "content": <nip04(<request>)>,
+    "tags": [["p", <remote pubkey>]],
+    "created_at": <unix timestamp in seconds>,
 }
 ```
 
-And the events the remote signer sends to the client have the following format:
+The `content` field is a JSON-RPC-like message that is [NIP-04](https://github.com/nostr-protocol/nips/blob/master/04.md) encrypted and has the following structure:
 
-```js
-  "pubkey": "<signer-key-hex>"
-  "kind": 24133,
-  "tags": [
-    ["p", "<client-key-hex>"]
-  ],
-  "content": "nip04_encrypted_json({id: <request-id>, result: <string>, error: <reason-string>})",
-  ...
+```json
+{
+    "id": <random_string>,
+    "method": <method_name>,
+    "params": [array_of_strings]
+}
 ```
 
-The signer key will always be the key of the user who controls the signer device.
-
-### Methods
-
-- **connect**
-  - params: [`pubkey`, `secret`]
-  - result: `"ack"`
-- **get_public_key**
-  - params: []
-  - result: `pubkey-hex`
-- **sign_event**
-  - params: [`event`]
-  - result: `json_string(event_with_pubkey_id_and_signature)`
-- **get_relays**
-  - params: []
-  - result: `json_string({[url: string]: {read: boolean, write: boolean}})`
-- **nip04_encrypt**
-  - params: [`third-party-pubkey`, `plaintext`]
-  - result: `nip04-ciphertext`
-- **nip04_decrypt**
-  - params: [`third-party-pubkey`, `nip04-ciphertext`]
-  - result: `plaintext`
-- **nip44_get_key**
-  - params: [`third-party-pubkey`]
-  - result: `nip44-conversation-key`
-- **nip44_encrypt**
-  - params: [`third-party-pubkey`, `plaintext`]
-  - result: `nip44-ciphertext`
-- **nip44_decrypt**
-  - params: [`third-party-pubkey`, `nip44-ciphertext`]
-  - result: `plaintext`
-- **ping**
-  - params: []
-  - result: `"pong"`
+- `id` is a random string that is a request ID. This same ID will be sent back in the response payload.
+- `method` is the name of the method/command (detailed below).
+- `params` is a positional array of string parameters.
+
+### Methods/Commands
+
+Each of the following are methods that the client sends to the remote signer.
+
+| Command                  | Params                                            | Result                                                                 |
+| ------------------------ | ------------------------------------------------- | ---------------------------------------------------------------------- |
+| `create_account`         | `[<username>, <domain>, <optional_email>]`        | `<newly_created_remote_pubkey>`                                        |
```

Comment:
per @pablof7z all these params are optional: https://github.com/nostr-protocol/nips/pull/976#discussion_r1445399476

I think it would be good to be able to discern that from the spec.

also @alexgleason [suggested](https://github.com/nostr-protocol/nips/pull/976#discussion_r1448119529) it should be 'name' to be consistent with nip05 
```suggestion
| `create_account`         | `[<optional-name>, <optional-domain>, <optional_email>]`        | `<newly_created_remote_pubkey>`                                        |
```

Reply:
I'm unclear as to how this would work if you didn't pass a username or domain option? Are you just then saying to a bunker, create me a new set of keys on your domain without a username?

Reply:
Yeah, I think that's perfectly valid, the client would simply not know which username was created but it can fetch it via nip05 since it has the pubkey

All these fields should be optional

Reply:
"the client would simply not know which username was created but it can fetch it via nip05 since it has the pubkey"

That's not great. NIP-05 queries can only be done by name (using the query string `?name=`). Services are not required to expose the full list of pubkeys to everybody.

Also if you don't know the domain name even that would be impossible to query.

`name` and `domain` cannot be optional.

Reply:
Agree, I don't think these can be optional.